added requires_openproject to Redmine::Plugin

cratz opened the pull request:
- method to check OpenProject core version from the rails3 plugins

Unfortunately we need this 'ugly RY'(TM) stuff until we moved everything to OpenProject namespace.

Commit:
added requires_openproject to Redmine::Plugin

- method to check OpenProject core version from the rails3 plugins

mfrister commented:
I'd like deprecation warnings when requires_redmine or requires_chiliproject is called. What do you think?

Commit:
added deprecation warning for ...

- Redmine::Plugin.requires_chiliproject() and
- Redmine::Plugin.requires_redmine()

cratz commented:
Good point, latest commit should show you what I think ;)

Commit:
removed requires_chiliproject and requires_redmine

completely
No migrated plugin should use the old methods and
the rails2 versions are not compatible anyway

Commit:
fixed plugin unit tests
